src/components/Project/ImageAnalysis/ImageDownloadMenu.vue

```diff
@@ -0,0 +1,64 @@
+<script setup>
+import { onMounted } from 'vue'
+import { useAlertsStore } from '@/stores/alerts'
+
+const props = defineProps(['image', 'tifUrl'])
+
+const emit = defineEmits(['analysisAction'])
+
+const alertStore = useAlertsStore()
+
+function downloadLink(link, filename, fileType) {
+  if(!link){
+    alertStore.setAlert('error', `No ${fileType} available for download`)
+    return
+  }
+  const a = document.createElement('a')
+  a.href = link
+  a.download = filename
+  document.body.appendChild(a)
+  a.click()
+  document.body.removeChild(a)
+}
+
+onMounted(() => {
+  const getTifInput = {
```

Comment:
It's not clear to me from the code but does this only call get tiff output once the user clicks the button? or does it call it when they enter the analysis view for an image? If its the former that is fine, but if its the later I don't think we want to generate the tiff everytime they just view the image in the analysis tab, we only want to have the backend generate the tiff on demand. I think the correct way would be to have the frontend call the backend to generate and return the tiff on click of the download tiff button.

Reply:
you made me have to do some freaky things to get this working but I learned something new